### PR TITLE
feat(icons): added `scan-qr-code` icon

### PR DESCRIPTION
**Pull request by vexkiddy**
## What is the purpose of this pull request?

- [x] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description
New scan qr-code icon

### Icon use case 
1. for indicating a scanner feature of qr-codes
2. to differentiate from the qr-code icon that doesn't indicate scanning

### Concept 

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [x] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [ ] I've based them on the following Lucide icons: 
- [ ] I've based them on the following design: 

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

File: icons/scan-qrcode.svg [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  width="24"
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M17 12v3a2 2 0 0 1-2 2h-3" />
```
Comment (jguddas) [inline icons/scan-qrcode.svg:12]: I would add a sharper corner radius there.
```suggestion
  <path d="M17 12v4a1 1 0 0 1-1 1h-4" /
```
<a title="Open lucide studio" target="_blank" href="https://lucide-studio.vercel.app/edit?value=%3Cpath+d=%22M17+12v4a1+1+0+0+1-1+1h-4%22+/%3E%3Cpath+d=%22M17+3h2a2+2+0+0+1+2+2v2%22+/%3E%3Cpath+d=%22M17+8V7%22+/%3E%3Cpath+d=%22M21+17v2a2+2+0+0+1-2+2h-2%22+/%3E%3Cpath+d=%22M3+7V5a2+2+0+0+1+2-2h2%22+/%3E%3Cpath+d=%22M7+17h.01%22+/%3E%3Cpath+d=%22M7+21H5a2+2+0+0+1-2-2v-2%22+/%3E%3Crect+x=%227%22+y=%227%22+width=%225%22+height=%225%22+rx=%221%22+/%3E&dialog=false&name=scan-qr-code">icons<br/>Open lucide studio</a>

**Review by jguddas — CHANGES_REQUESTED** [review]
The icon should be renamed to `scan-qr-code`.

**Comment by vexkiddy** [comment]
hiya @jguddas think i've approved everything.. do i need do anything else ?

**Comment by vexkiddy** [comment]
@jguddas just nudging this :D

**Review by jguddas — APPROVED** [review]

File: icons/scan-qr-code.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "barcode",
+    "scan",
+    "qrcode",
+    "url",
+    "information",
+    "digital",
+    "scanner"
+  ],
+  "categories": [
+    "development"
```
Comment (ericfennis) [inline icons/scan-qr-code.json:17]: Not sure if this is the correct category. This not really associated with development in general
I think these categories fit better:
```suggestion
    "account",
    "shopping",
    "devices",
    "security"
```

**Review by ericfennis — APPROVED** [review]

**Pull request merged**